### PR TITLE
fix submenu focus

### PR DESCRIPTION
**Pull request by SpaNb4**
Fixes the following issues:
1. The focus does not shift back when reopening the menu when a submenu item is active

https://github.com/react-component/menu/assets/69504277/d0d9493f-2682-4099-8d3c-e85fc3169098

2. (preact specific) When a sub-menu opens, I can't use the arrow keys to navigate through it

https://github.com/react-component/menu/assets/69504277/afdb5c8a-9d65-41d7-ace9-e8563fa41c06

Note: I use the antd Dropdown component in the videos above

### CODE REVIEW

**Comment by yoyo837** [comment]
Please add some test cases to cover this.

**Comment by SpaNb4** [comment]
> Please add some test cases to cover this.

I wanted to add some tests, but then I realized that rc-menu doesn't have an `autoFocus` prop that antd uses. So I'm not sure if it's possible to reproduce this behavior.
And speaking of the second issue, I don't think it's possible to test it, since you need to use preact in your tests

File: src/SubMenu/index.tsx [hunk 1]
```diff
@@ -137,6 +137,12 @@ const InternalSubMenu = (props: SubMenuProps) => {
     onTitleMouseLeave,
   );
 
+  React.useEffect(() => {
+    if (elementRef.current && activeKey === eventKey) {
+      elementRef.current.focus();
+    }
+  });
```
Comment (zombieJ) [inline src/SubMenu/index.tsx:144]: Seems this code makes dead loop which makes test OOM.

Comment (SpaNb4) [reply src/SubMenu/index.tsx]: Removed that part and pushed another fix. The real issue was that the submenu item is `div` element, not `li`
image

**Review by yoyo837 — APPROVED** [review]

**Comment by zombieJ** [comment]
cc @MadCcc

**Comment by MadCcc** [comment]
Is this issue related?
https://github.com/ant-design/ant-design/issues/43267

**Comment by yoyo837** [comment]
> Is this issue related? [ant-design/ant-design#43267](https://github.com/ant-design/ant-design/issues/43267)

Yes, I think.

File: src/SubMenu/index.tsx [hunk 2]
```diff
@@ -1,26 +1,26 @@
...
+import Icon from '../Icon';
+import type { MenuInfo, SubMenuType } from '../interface';
+import { parseChildren } from '../utils/commonUtil';
+import { warnItemProp } from '../utils/warnUtil';
+import InlineSubMenuList from './InlineSubMenuList';
+import PopupTrigger from './PopupTrigger';
+import SubMenuList from './SubMenuList';
 
 export interface SubMenuProps
   extends Omit<SubMenuType, 'key' | 'children' | 'label'> {
```
Comment (MadCcc) [inline src/SubMenu/index.tsx:26]: Change on this file should be reverted since nothing except order changed.

**Comment by MadCcc** [comment]
> I wanted to add some tests, but then I realized that rc-menu doesn't have an `autoFocus` prop that antd uses. So I'm not sure if it's possible to reproduce this behavior.

Test for the first problem should be provided, you can trigger focus manually in test case to mock the scene.

**Comment by SpaNb4** [comment]
> Is this issue related? [ant-design/ant-design#43267](https://github.com/ant-design/ant-design/issues/43267)

No, this PR is not relevant to the above issue

**Comment by SpaNb4** [comment]
> > I wanted to add some tests, but then I realized that rc-menu doesn't have an `autoFocus` prop that antd uses. So I'm not sure if it's possible to reproduce this behavior.
> 
> Test for the first problem should be provided, you can trigger focus manually in test case to mock the scene.

Ok, I'll try

**Comment by yoyo837** [comment]
Now it conflicts.

**Comment by SpaNb4** [comment]
Closing this since I can no longer reproduce the first issue

**Pull request closed without merging**